xonsh/main.py

```diff
@@ -252,10 +253,58 @@ def premain(argv=None):
     return args
 
 
+def _failback_to_other_shells(argv):
+    args = None
+    try:
+        args = premain(argv)
+    except Exception:
+        pass
+    # only failback for interactive shell; if we cannot tell, treat it
+    # as an interactive one for safe.
+    if hasattr(args, 'mode') and args.mode != XonshMode.interactive:
+        sys.exit(1)
```

Comment:
That is a fair point. But maybe we should just rely on the return below and and call sys.exit().  sys.exit is problematic if you want to call the main function programtically - which is why I suggested reraising. I guess I think we should just return then.  Really, I just want to avoid the sys.exit if we can.